File: framework/ui/src/io/intino/alexandria/ui/displays/notifiers/ComponentNotifier.java
package io.intino.alexandria.ui.displays.notifiers;

import io.intino.alexandria.http.pushservice.MessageCarrier;
import io.intino.alexandria.ui.displays.Display;
import io.intino.alexandria.ui.displays.UserMessage;

public class ComponentNotifier extends DisplayNotifier {

	public ComponentNotifier(Display display, MessageCarrier carrier) {
		super(display, carrier);
	}

	public void refreshLoading(Boolean value) {
		putToDisplay("refreshLoading", "v", value);
	}

	public void userMessage(UserMessage value) {
		putToDisplay("userMessage", "v", value);
	}

	public void hideUserMessage() {
		putToDisplay("hideUserMessage");
	}

	public void refreshVisibility(boolean value) {
		putToDisplay("refreshVisibility", "v", value);
	}

	public void refreshColor(String color) {
		putToDisplay("refreshColor", "v", color);
	}

	public void refreshFormat(String format) {
		putToDisplay("refreshFormat", "v", format);
	}

}

File: framework/ui-elements/src/displays/notifiers/ComponentNotifier.js
import Notifier from "../../../gen/displays/notifiers/Notifier";

export default class ComponentNotifier extends Notifier {

	constructor(element) {
		super(element);
		this.setup();
	};

	setup() {
		super.setup();
		if (this.element == null || this.pushLinked) return;
		this.when("refreshLoading").toSelf().execute((parameters) => this.element.refreshLoading(parameters.v));
		this.when("refreshVisibility").toSelf().execute((parameters) => this.element.refreshVisibility(parameters.v));
		this.when("refreshColor").toSelf().execute((parameters) => this.element.refreshColor(parameters.v));
		this.when("userMessage").toSelf().execute((parameters) => this.element.userMessage(parameters.v));
		this.pushLinked = true;
	};

}